amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/LeaseCleanupManager.java

```diff
@@ -81,12 +81,16 @@
      * {@link LeaseCleanupManager#leaseCleanupIntervalMillis}
      */
     public void start() {
-        log.debug("Starting lease cleanup thread.");
-        completedLeaseStopwatch.start();
-        garbageLeaseStopwatch.start();
-
-        deletionThreadPool.scheduleAtFixedRate(new LeaseCleanupThread(), INITIAL_DELAY, leaseCleanupIntervalMillis,
-                TimeUnit.MILLISECONDS);
+        if (!isRunning) {
```

Comment:
Given that we are checking for running here, we can actually need not check it outside from scheduler